macro/src/snapshots/diplomat__tests__traits.snap

```diff
@@ -0,0 +1,87 @@
+---
+mod ffi {
+    #[repr(C)]
+    pub struct TestingStruct {
+        x: i32,
+        y: i32,
+    }
+    pub trait TesterTrait {
+        fn test_trait_fn(&self, x: i32) -> i32;
+        fn test_void_trait_fn(&self);
+        fn test_struct_trait_fn(&self, s: TestingStruct) -> i32;
+        fn test_slice_trait_fn(&self, s: &[u8]) -> i32;
+    }
+    #[repr(C)]
+    pub struct Wrapper {
+        cant_be_empty: bool,
+    }
+    impl Wrapper {
+        pub fn test_with_trait(t: impl TesterTrait, x: i32) -> i32 {
+            t.test_void_trait_fn();
+            t.test_trait_fn(x)
+        }
+        pub fn test_trait_with_struct(t: impl TesterTrait) -> i32 {
+            let arg = TestingStruct { x: 1, y: 5 };
+            t.test_struct_trait_fn(arg)
+        }
+    }
+    use core::ffi::c_void;
+    use diplomat_runtime::*;
+    #[no_mangle]
+    extern "C" fn Wrapper_test_with_trait(t: DiplomatTraitStruct_TesterTrait, x: i32) -> i32 {
+        Wrapper::test_with_trait(t, x)
+    }
+    #[no_mangle]
+    extern "C" fn Wrapper_test_trait_with_struct(t: DiplomatTraitStruct_TesterTrait) -> i32 {
+        Wrapper::test_trait_with_struct(t)
+    }
+    #[repr(C)]
+    pub struct TesterTrait_VTable {
+        pub destructor: Option<unsafe extern "C" fn(*const c_void)>,
+        pub size: usize,
+        pub alignment: usize,
+        pub run_test_trait_fn_callback: unsafe extern "C" fn(*const c_void, i32) -> i32,
+        pub run_test_void_trait_fn_callback: unsafe extern "C" fn(*const c_void),
+        pub run_test_struct_trait_fn_callback:
+            unsafe extern "C" fn(*const c_void, TestingStruct) -> i32,
+        pub run_test_slice_trait_fn_callback:
+            unsafe extern "C" fn(*const c_void, diplomat_runtime::DiplomatSlice<u8>) -> i32,
+    }
+    #[repr(C)]
+    pub struct DiplomatTraitStruct_TesterTrait {
```

Comment:
thought: an annoying thing about this is that someone wishing to use TesterTrait in a different module will have to manually import this type, which we haven't typically needed in the macro. It's tricky to avoid though, the main thing I can think of is scouring the imports for `TesterTrait` and tacking on an additional import of the trait struct.

Reply:
Oh yeah that's a good point. I guess we can add a note about that in the docs; I can't think of how to avoid it